agent/hcp/telemetry_provider.go

```diff
@@ -67,21 +85,47 @@ func defaultDisabledCfg() *dynamicConfig {
 	}
 }
 
+// httpCfg is a set of configurable settings for the HTTP client used to export metrics
+type httpCfg struct {
+	header *http.Header
+	client *retryablehttp.Client
+}
+
+type HCPProviderCfg struct {
+	HCPClient client.Client
+	HCPConfig config.CloudConfigurer
+}
+
 // NewHCPProvider initializes and starts a HCP Telemetry provider.
-func NewHCPProvider(ctx context.Context, hcpClient client.Client) *hcpProviderImpl {
+func NewHCPProvider(ctx context.Context) *hcpProviderImpl {
 	h := &hcpProviderImpl{
 		// Initialize with default config values.
-		cfg:       defaultDisabledCfg(),
-		hcpClient: hcpClient,
+		cfg:     defaultDisabledCfg(),
+		httpCfg: &httpCfg{},
+		logger:  hclog.FromContext(ctx),
+	}
+
+	return h
+}
+
+// Run continously checks for updates to the telemetry configuration by making a request to HCP.
+func (h *hcpProviderImpl) Run(ctx context.Context, c *HCPProviderCfg) error {
+	// Update the provider with the HCP configurations
+	h.hcpClient = c.HCPClient
```

Comment:
The HCPClient is more associated with the dynamic config, so I think it would be better off adding back the GetHCPClient/SetHCPClient methods.

Or alternatively, it might be easier in the long term to track whether the provider is already running and only allow it to update and run if it's not already running. That way, we don't always have to call `GetHCPClient` when we need to use the client. And for our current use cases, we won't be updating the HCP client once it starts running.

Reply:
I went with checking if it's running before updating the HCP client since I thought that this check is useful anyway to keep us from spawning multiple running goroutines!